src/bootstrap/src/utils/exec.rs

```diff
@@ -70,25 +84,36 @@ pub struct BootstrapCommand {
     // This field makes sure that each command is executed (or disarmed) before it is dropped,
     // to avoid forgetting to execute a command.
     drop_bomb: DropBomb,
+    should_cache: bool,
 }
 
 impl<'a> BootstrapCommand {
     #[track_caller]
     pub fn new<S: AsRef<OsStr>>(program: S) -> Self {
-        Command::new(program).into()
+        Self { should_cache: true, ..Command::new(program).into() }
```

Comment:
`should_cache` should now already be set to `true` in the `From` impl from `StdCommand`.